idaes/core/util/model_diagnostics.py

```diff
@@ -3887,6 +3930,51 @@ def b_rule(b, i):
     return ill_cond
 
 
+def compute_infeasibility_explanation(
```

Comment:
Does this need to be done as a separate method, or could it just be written directly into the toolbox class?

Reply:
I agree -- I couldn't decide at what level things should be integrated. Implemented this suggestion in [40670e4](https://github.com/IDAES/idaes-pse/pull/1409/commits/40670e4dc0d66a766d87d4f70fb7820ecbf30965).